src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplate.java

```diff
@@ -81,6 +83,8 @@
 
     private String resourceLimitMemory;
 
+    private String restartPolicy = DEFAULT_POD_RESTART_POLICY;
```

Comment:
If I am not mistaken, this will not work when this is filled from an old config that does not have `restartPolicy`.

You need to implement `private Object readResolve()` to fill `restartPolicy` in case it is `null`. See https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility:

> Persistence is XML-based, and when the data is read back from disk and XML doesn't contain data for a particular field, the existing field value is left as-is. "As-is" normally means the JVM default value (0, null, etc.), because the persistence doesn't invoke your constructor to create your object, like Java serialization.